### PR TITLE
[ANE-1070] Pass licenseScanPathFilters to lernie

### PR DESCRIPTION
**Pull request by james-fossa**
# Overview

Passes `LicenseScanPathFilters` to lernie to give `only` and `exclude` globs to lernie pursuant to [this PR](https://github.com/fossas/lernie/pull/125).

## Acceptance criteria

Passes the new arguments to Lernie.

## Testing plan

- `rm ./vendor-bins/*`
- `./vendor
- Create a directory structure like that below (`mkdir -p a/b b c/b d/e/b && echo test | tee a/b/license.txt b/license.txt b/not-license.txt c/b/license.txt d/e/b/license.txt > /dev/null && touch a/b/pom.xml b/pom.xml d/e/b/pom.xml` works)
- Use a `.fossa.yml` like that below.
- Run `fossa analyze --output` on the dir. You should see results for every `license.txt` but not `not-license.txt`
- Uncomment the exclude line and repeat. You should see no results.

### Tree Structure
```
├── a
│   └── b
│       ├── license.txt
│       └── pom.xml
├── b
│   ├── license.txt
│   ├── not-license.txt
│   └── pom.xml
├── c
│   └── b
│       └── license.txt
└── d
    └── e
        └── b
            ├── license.txt
            └── pom.xml
```

### Fossa Yaml

```yaml
version: 3

server:: https://app.fossa.com
apiKey: REDACTED

project:
  locator: custom+1/jclemer/ane-1070

customLicenseSearch:
  - matchCriteria: test
    name: "Test ANE-1070"

vendoredDependencies:
dependencies.md#path-filtering
  licenseScanPathFilters:
    only:
      - "**/*.txt"
    exclude:
      - "**/not-license.txt"
      # To test that exclude works on the directory level, un-comment the next line:
      # - "**/b/*.txt"
customLicenseSearch:
   - matchCriteria: test
     name: "Proprietary License"
```
### Output
Should include

#### With just the `not-license.txt` Exclusion
```
* Custom-License Search: succeeded
  ** Proprietary License - /Users/jclemer/testcli/a/b/license.txt (lines 1-1)
  ** Proprietary License - /Users/jclemer/testcli/c/b/license.txt (lines 1-1)
  ** Proprietary License - /Users/jclemer/testcli/b/license.txt (lines 1-1)
  ** Proprietary License - /Users/jclemer/testcli/d/e/b/license.txt (lines 1-1)
```
#### With Exclusions
```
* Custom-License Search: succeeded
  ** No results found
```

## Risks

I'm pretty confident the glob library we're using in lernie is semantically compatible with what's going on in themis but there could always be an edge case somewhere.

## Metrics

N/A

## References

- [ANE-1070](https://fossa.atlassian.net/browse/ANE-1070)

## Checklist

- [x] I added tests for this PR's change (or explained in the PR description why tests don't make sense).
- [x] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [x] If this PR added docs, I added links as appropriate to the user manual's ToC in `docs/README.ms` and gave consideration to how discoverable or not my documentation is.
- [x] If this change is externally visible, I updated `Changelog.md`. If this PR did not mark a release, I added my changes into an `## Unreleased` section at the top.
- [x] If I made changes to `.fossa.yml` or `fossa-deps.{json.yml}`, I updated `docs/references/files/*.schema.json` AND I have updated example files used by `fossa init` command. You may also need to update these if you have added/removed new dependency type (e.g. `pip`) or analysis target type (e.g. `poetry`).
- [x] If I made changes to a subcommand's options, I updated `docs/references/subcommands/<subcommand>.md`.

[ANE-1070]: https://fossa.atlassian.net/browse/ANE-1070?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: test/App/Fossa/LernieSpec.hs [hunk 1]
```diff
@@ -391,10 +402,31 @@ spec = do
...
+      result <- ignoreDebug . withoutTelemetry $ analyzeWithLernie scanDir Nothing grepOptions Nothing
+      case result of
+        Nothing -> expectationFailure' "analyzeWithLernie should not return Nothing"
+        Just _ -> pure ()
+
+    it' "should apply licenseScanPathFilters correctly" $ do
+      let filters =
+            LicenseScanPathFilters
+              { licenseScanPathFiltersOnly = [GlobFilter "**/*one.txt"]
+              , licenseScanPathFiltersExclude = [GlobFilter "**/*something.txt"]
```
Comment (spatten) [inline test/App/Fossa/LernieSpec.hs:423]: [nit] This isn't testing the exclude filter, as the `only` filter is already filtering out something.txt. It's probably worth adding a test of just the exclude filter too.

**Review by spatten — APPROVED** [review]
Looks great! I left one small comment on the test, but nothing blocking
